### PR TITLE
qt6-qtcreator: new port

### PR DESCRIPTION
**Pull request by rue-ryuzaki**
#### Description

Portfile has been updated for the new Qt Creator version:
- build with cmake
- use clang-17 from macports instead of host clang
- remove unused patch clang16_Wenum_constexpr_conversion
- adapt patch no_qtwebengine for cmake
- use c++ standard 20 (defined in CMakeLists.txt)
- update Qt Creator.app build directory for destroot
- remove move libexec/QtProject to Resources (must be already there)

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 15.1 24B83 x86_64
Xcode 16.1 16B40

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by reneeotten** [comment]
this doesn't seem right. You commented on another PR that the version you're updating the port here uses Qt6; so that should certainly not be called "qt5-...". 
Please upsat the "qt5-qtcreator" port to the latest version for Qt5 and then you can make a new port for Qt6. In this case let's create a new port instead of a subport as the build system has changed.

**Comment by rue-ryuzaki** [comment]
@reneeotten Hi, I'm not sure I did everything right.
While I was porting the port file, I discovered that the assembly is carried out by the host compiler. Accordingly, because of this, a build error occurred with clang16 (although clang12 was specified in the Portfile). I tried to fix it for qt5-qtcreator but there were problems with it. Also now there is a problem with building qt5-qtcreator on older macos systems (for example 10.13) due to the qbs update, so there is still a fix in its port. And it will take an indefinite amount of time.

This PR no longer a port update. So, it will be new port, port fix or can be closed (and there will be separate PR)?

I also update the qt creator version from 14.0.2 to 15.0.0 (it was released recently)

**Comment by reneeotten** [comment]
I will try and take a look at this in the weekend.

File: devel/qt6-qtcreator/Portfile [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+revision            0
+categories          devel aqua
+# from qt6 PortGroup:
+# qt 6.8 - darwin >= 20
+# qt 6.4 - darwin >= 18
+platforms           {darwin >= 18}
+# from https://blog.qt.io/blog/2016/01/13/new-agreement-with-the-kde-free-qt-foundation/
+# The GPL license comes with two exceptions that ensure that there are no license restrictions on generated code, and that bridging to 3rd party code is still possible.
+license             GPL-3
+maintainers         {mcalhoun @MarcusCalhoun-Lopez} openmaintainer
```
Comment (markemer) [inline devel/qt6-qtcreator/Portfile:19]: I know @MarcusCalhoun-Lopez was the qt/qt5-creator maintainer, but this is new, we probably shouldn't sign him up.

**Comment by markemer** [comment]
This looks good, I'm ok merging it, but I'd like to get a second opinion from @reneeotten who's kinda the new qt maintainer/expert around here.

**Pull request merged**